### PR TITLE
8262492: Add cast nodes between float types implementation for Arm SVE

### PR DESCRIPTION
**Pull request by Wanghuang-Huawei**
Add cast nodes like castf2d castd2f

* add `sve_fcvt`
* reuse macro `VECTOR_CAST_I2F_NARROW1` `VECTOR_CAST_I2F_EXTEND1` (Should we change these macro names?)

other issues :
* precision loss

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Change must be properly reviewed

### Issue
 * [JDK-8262492](https://bugs.openjdk.java.net/browse/JDK-8262492): Add cast nodes between float types implementation for Arm SVE

### Reviewers
 * [Ningsheng Jian](https://openjdk.java.net/census#njian) (@nsjian - Committer)

### Contributors
 * Wang Huang `<whuang@openjdk.org>`
 * He Xuejin `<hexuejin2@huawei.com>`
 * Ai Jiaming `<aijiaming1@huawei.com>`

### Download
`$ git fetch https://git.openjdk.java.net/panama-vector pull/41/head:pull/41`
`$ git checkout pull/41`

### CODE REVIEW

File: src/hotspot/cpu/aarch64/assembler_aarch64.hpp [hunk 1]
```diff
@@ -3403,6 +3403,53 @@ void mvnw(Register Rd, Register Rm,
...
+        opc = 0b11;
+        opc2 = 0b00;
+      } else if (T_dst == S) {
+        opc = 0b11;
+        opc2 = 0b10;
+      }
+    }
+  }
+
+// SVE floating-point convert precision(predicated)
```
Comment (nsjian) [inline src/hotspot/cpu/aarch64/assembler_aarch64.hpp:3438]: space between precision and '(' ?

File: src/hotspot/cpu/aarch64/assembler_aarch64.hpp [hunk 2]
```diff
@@ -3403,6 +3403,53 @@ void mvnw(Register Rd, Register Rm,
...
+            FloatRegister Zn, SIMD_RegVariant T_src) {                  \
+    starti;                                                             \
+    unsigned opc, opc2;                                                 \
+    encode_fcvt_T(T_src, T_dst, opc, opc2);                             \
+    f(0b01100101, 31, 24), f(opc, 23, 22), f(0b0010, 21, 18);           \
+    f(opc2, 17, 16), f(0b101, 15, 13);                                  \
+    pgrf(Pg, 10), rf(Zn, 5), rf(Zd, 0);                                 \
+  }
+
+  INSN(sve_fcvt);
```
Comment (nsjian) [inline src/hotspot/cpu/aarch64/assembler_aarch64.hpp:3450]: Since there's only one use of the INSN macro, I think it's no need to define a macro here.

**Review by nsjian — COMMENTED** [review]
> * reuse macro `VECTOR_CAST_I2F_NARROW1` `VECTOR_CAST_I2F_EXTEND1` (Should we change these macro names?)

I think it would be better to change the macro name.

File: src/hotspot/cpu/aarch64/assembler_aarch64.hpp (hunk 1 above)
Comment (Wanghuang-Huawei) [reply src/hotspot/cpu/aarch64/assembler_aarch64.hpp:3438]: OK. I will change that.

File: src/hotspot/cpu/aarch64/assembler_aarch64.hpp (hunk 2 above)
Comment (Wanghuang-Huawei) [reply src/hotspot/cpu/aarch64/assembler_aarch64.hpp:3450]: OK. I will clear that.

**Review by nsjian — APPROVED** [review]
Looks good to me.

**Pull request closed without merging**